nixos/modules/services/misc/ollama.nix

```
@@ -3,12 +3,19 @@ let
   inherit (lib) types;
 
   cfg = config.services.ollama;
-  ollamaPackage = cfg.package.override {
+
+  stableOverrides = {
     inherit (cfg) acceleration;
     linuxPackages = config.boot.kernelPackages // {
       nvidia_x11 = config.hardware.nvidia.package;
     };
   };
+  unstableOverrides = {
+    inherit (cfg) acceleration;
+  };
```

Potential alternatives to 1) maintaining two attrsets, 2) testing for passthru attributes:

```suggestion
  overrides = {
    inherit (cfg) acceleration;
  } // lib.optionalAttrs (lib.versionAtLeast lib.version "24.11pre") {
    linuxPackages = config.boot.kernelPackages // {
      nvidia_x11 = config.hardware.nvidia.package;
    };
  };
```

(I probably messed up the formatting)

---

The passthru solution is maybe better for mixing packages from different nixpkgs revisions

---

As far as I can tell this wouldn't work at all because the problem comes from using the unstable package in the stable module, which "expects" the stable package that does still use `linuxPackages`. The `linuxPackages` override is useful for the stable package, but breaks the unstable package. Therefore, whether to include `linuxPackages` in the override needs to be dynamic based on whether the package used is stable or unstable, or more specifically whether it does or does not have `linuxPackages` as an input.

---

You did catch that this tests passthru args, which is not intended. I meant to test the function inputs when the package function is called by `pkgs.callPackage`.

Do you remember the name of the function inputs attrset? I seem to remember something like `package.drvAttrs`, but I can't remember well, and I'm having trouble finding documentation about it online.

---

Not sure. You (often) can inspect callPackage arguments by abusing `lib.functionArgs pkg.override`